lib/plugins/filter/meta_generator.js

```diff
@@ -1,21 +1,17 @@
 'use strict';
 
-let cheerio;
-const hexoGeneratorTag = '<meta name="generator" content="Hexo %s" />';
-
 function hexoMetaGeneratorInject(data) {
   const { config } = this;
-  if (!config.meta_generator) return;
-
-  if (!cheerio) cheerio = require('cheerio');
-  const $ = cheerio.load(data, {decodeEntities: false});
+  if (!config.meta_generator ||
+  data.includes('<meta name="generator" content="Hexo')) return;
 
-  if (!($('meta[name="generator"]').length > 0) &&
-  $('head').contents().length > 0) {
-    $('head').prepend(hexoGeneratorTag.replace('%s', this.version));
+  const hexoGeneratorTag = '<meta name="generator" content="Hexo %s" />'
+    .replace('%s', this.version);
 
-    return $.html();
-  }
+  // meta generator should not be prepended empty <head></head>
+  // see https://github.com/hexojs/hexo/pull/3315
+  return data.replace('<head></head>', '')
```

Comment:
I need help replacing the line to a regex.
`/(?!<head>)<\/head>/g` doesn't work.

``` js
const headTag = '<head><span></span></head><head></head>'

console.log(headTag.replace(/(?!<head>)<\/head>/g, 'ccc</head>'))
// <head><span></span>ccc</head><head>ccc</head>
// expected <head><span></span>ccc</head><head></head>
```

---

```
headTag.replace(/<\/head>/, 'ccc</head>')
```
without `/g` global modifier, meaning replace only the first occurrence, may work. But I suspect there may be empty `<head></head>` in the actual head ([thanks to cheerio](https://github.com/hexojs/hexo/pull/3315)).

Reply:
https://github.com/gohugoio/hugo/blob/master/transform/metainject/hugogenerator.go
Maybe we could learn something from hugo.

Reply:
Hugo doesn't need to deal with [cheerio's quirks](https://github.com/hexojs/hexo/pull/3315) :frowning: